app/view/main/Map.js

```diff
@@ -177,6 +192,119 @@ Ext.define('CpsiMapview.view.main.Map', {
             });
         }
 
+        if (me.enablePermalink) {
+            if (window.location.hash !== '') {
+                // try to restore center, zoom-level and rotation from the URL
+                me.applyState(me.readPermalink(window.location.hash));
+            }
+
+            me.registerPermalinkEvents();
+        }
+
         Ext.GlobalEvents.fireEvent('cmv-mapready', me);
+    },
+
+    /**
+     * Registers the events to have a permalink synchronization.
+     *
+     * @private
+     */
+    registerPermalinkEvents: function () {
+        var me = this;
+
+        // update permalink when visible map state changes
+        me.olMap.on('moveend', me.updatePermalink, me);
+
+        // restore the view state when navigating through the history, see
+        // https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onpopstate
+        window.addEventListener('popstate', function(event) {
+            if (event.state === null) {
+                return;
+            }
+            me.olMap.getView().setCenter(event.state.center);
+            me.olMap.getView().setZoom(event.state.zoom);
+            me.olMap.getView().setRotation(event.state.rotation);
+            me.shouldUpdatePermalink = false;
+        });
+    },
+
+    /**
+     * Returns the current map state (center, zoom, rotation)
+     * @return {Object} The map state object
+     */
+    getState: function () {
+        var me = this;
+        var view = me.olMap.getView();
+        return {
+            zoom: view.getZoom(),
+            center: view.getCenter(),
+            rotation: view.getRotation()
+        };
+    },
+
+    /**
+     * Applies the map state (center, zoom, rotation) to this map.
+     *
+     * @param  {Object} mapState The map state object
+     */
+    applyState: function (mapState) {
+        var me = this;
+
+        me.olMap.getView().setCenter(mapState.center);
```

Comment:
`mapState` could be an empty object / null here if the permalink could not be read, or am I wrong? Could you please add a null check?

Reply:
Yep, added check in 5aac0ec